spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/GatewayApiController.java

```diff
@@ -51,4 +68,30 @@ public ApiKey findApiKeyEntityByApiKeyValue(@RequestParam("keyValue") String api
     public ServiceApi findServiceByApiKeyValue(@RequestParam("apiKeyValue") String apiKeyValue) {
         return serviceApiRepository.findByApiKeyValue(apiKeyValue);
     }
+
+    @GetMapping( "/api/gateways/api-keys")
+    public ResponseEntity<List<ApiKey>> findApiKeyListByGatewayApiKey(@RequestParam("gateway-api-key") String gatewayApiKey) {
+        log.debug("The gateway with the ID {} requests its latest API key list.", gatewayApiKey);
+        try {
+            Proxy proxy = proxyRepository.findById(gatewayApiKey);
```

Comment:
This is the origin of the 'API key' part:
```public static final String GATEWAY_API_KEY_REQUEST_PARAMETER_NAME = "gateway-api-key";```

I think this is confusing and it would make sense to rename ```GATEWAY_API_KEY_REQUEST_PARAMETER_NAME``` to something like ```GATEWAY_ID_PARAMETER_NAME```. The new value of that field could simply be "gateway-id".

Reply:
In general I agree with you, however this would "break" existing running CoatRack gateways, as they use the parameter `gateway-api-key` to pass their ID.

I would suggest to discuss this general question in a separate issue and for now just add an explanatory comment into the code.

Reply:
I changed the arguments name to ```gatewayIdAndApiKey```.

I will create a separate issue on that matter.